### ISSUE TITLE
Java generator generates model classes with dashes in the name

### ISSUE THREAD

**Issue by ePaul** [swagger-api/swagger-codegen#1142]
**Summary**
When a `definitions` name contains a dash (`-`), the model class generated from this is named with a dash, too. This is not valid Java code.

This happens with current master of swagger-codegen, and I think it happened also with version 2.1.3.

**Reproducible by**

bug-trigger.yaml:
```
swagger: '2.0'
info:
  title: Swagger Codegen bug trigger
  description: |
     This demonstrates a bug in swagger-codegen.
  version: 0.0.1
basePath: /api
definitions:
  example-parent:
    type: object
    properties:
      foo:
        type: string
paths: {}
```
command line:

    java -jar /home/paulo/.m2/repository/io/swagger/swagger-codegen-cli/2.1.4-SNAPSHOT/swagger-codegen-cli-2.1.4-SNAPSHOT.jar generate -i src/main/resources/api/bug-trigger.yaml -l java -o target/generated-sources/swagger-codegen/

generated Example-parent.java (just the start):

```
package io.swagger.client.model;




import io.swagger.annotations.*;
import com.fasterxml.jackson.annotation.JsonProperty;


@ApiModel(description = "")
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2015-08-27T17:21:13.173+02:00")
public class Example-parent   {
```

Due to the dash in file name and class name, this can't be compiled.

**Expected behavior**
The names of classes should be sanitized so the generated identifiers are actually valid Java identifiers.

**Workaround**
Use only names which are already valid identifiers in your target language, e.g. without any dashes when using Java.

**Comment by ePaul** [comment]
#939 (Fixed with #940) had a similar issue, but for dashes in paths, not dashes in model definitions.

I guess similar problems could occur for predefined responses or parameters. Those should all be treated the same.

**Comment by ePaul** [comment]
By merging #1164 + #1163 this is now solved for models and spring-mvc API entry points.
I'm not sure whether this occurs somewhere else, then a similar approach can be used.

I'm therefore closing this issue.
